src/main/kotlin/io/zerobase/smarttracing/GraphDao.kt

```diff
@@ -216,4 +223,87 @@ class GraphDao(private val driver: Driver, val phoneUtil: PhoneNumberUtil) {
             }
         }
     }
+
+    /**
+     * Creates a user node and links to device id.
+     *
+     * @param name name of the user.
+     * @param phone phone of the user.
+     * @param email email of the user.
+     * @param id id of the device used to create it.
+     *
+     * @returns id of the user.
+     */
+    fun createUser(name: String?, phone: String?, email: String?, id: String): UserId? {
```

Comment:
can this function actually return `null` or does it throw errors when there is a problem? if it can't actually return null, don't make the signature nullable

Reply:
throws errors, but i will need to use !! to make it not nullable, because the type returned from the session is a nullable type